src/lib/core/theming/theme-check.ts

```diff
@@ -0,0 +1,40 @@
+import {NgModule, isDevMode} from '@angular/core';
+
+/** Whether the theme presence has already been checked. */
+let hasBeenChecked = false;
+
+/**
+ * Module that verifies that the user has loaded the core theming file,
+ * without which most Material module won't work as expected.
+ *
+ * Note on testing methodology: A more efficient way to check for the theme
+ * would be to loop through the `document.styleSheets`, however most browsers
+ * don't expose the stylesheet rules, if the file was loaded from another domain.
+ * This would trigger false positives if the theme is being loaded from a CDN.
+ *
+ * @docs-private
+ */
+@NgModule()
+export class MdThemeCheckModule {
+  constructor() {
+    if (hasBeenChecked || typeof document === 'undefined' || !isDevMode()) {
+      return;
+    }
+
+    let testElement = document.createElement('div');
+
+    testElement.classList.add('md-theme-loaded-marker');
+    document.body.appendChild(testElement);
+
+    if (getComputedStyle(testElement).display !== 'none') {
```

Comment:
I think that it may end up being slower. Both `getComputedStyle` and `offsetWidth` cause a reflow, however we only need to call `getComputedStyle` once. As an aside, I tried doing some quick measurements between my initial approach (looping through the `document.styleSheets`) and the `getComputedStyle` approach. Both ended up taking about the same time on Chrome, but the latter was a bit faster in IE.